packages/ruleset/src/utils/schema-finding-utils.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright 2023 IBM Corporation.
+ * SPDX-License-Identifier: Apache2.0
+ */
+
+const { isObject } = require('@ibm-cloud/openapi-ruleset-utilities');
+const getResponseCodes = require('./get-response-codes');
+
+const {
+  isJsonMimeType,
+  isJsonPatchMimeType,
+  isMergePatchMimeType,
+} = require('./mimetype-utils');
+
+/**
+ * Takes an operation object/path combo and finds the JSON success
+ * response (200) schema, if it exists. We're only interested in
+ * JSON content and this is to be used for exclusively GET operations
+ * (individual "gets" or "lists") so we only look at 200 status codes.
+ * 
```

Comment:
You bring up a good point - it does look like the Handbook is not even so strict everywhere and we don't have rules verifying that behavior. It may be best to be more flexible. Though, in the case of determining a canonical schema, I'm not sure what other status code would make sense.

I can make this more flexible, or perhaps the answer is to add a new rule that verifies the existence of 200 responses for "get" and "lists" on resource-oriented paths. Or perhaps both 🙂 What do you think?

Reply:
I think for this rule it might be good to be more tolerant of "incorrect" success status codes...  perhaps just look for the first response that matches against `2\d\d` and defines a schema?
Separately, we might also want to have a rule that makes sure get operations use the correct status codes, although I think we might have something in that area already IIRC.

Reply:
This is in a utility so it'll apply to more rules than this current one, but I agree it should be more flexible. I can make that change.

We do have a rule in this area but it's not strict on GET operations - I believe it's fine with any success code. That may be correct - e.g. I think it's acceptable to use a 204 for a GET if the resource has no body representation. So perhaps we just make this more flexible and leave the other rules as is